### PR TITLE
Create a base TestCaseEditor for rulesets to derive

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] #2205

I have started work on mania to make sure that the code is re-usable/to test the coding overhead. Eventually we're going to want to do this for every ruleset.

### CODE REVIEW

File: osu.Game/Tests/Visual/TestCaseEditor.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Game.Rulesets;
+using osu.Game.Screens.Edit;
+using osu.Game.Screens.Edit.Screens;
+using osu.Game.Tests.Beatmaps;
+
+namespace osu.Game.Tests.Visual
+{
+    public abstract class TestCaseEditor : OsuTestCase
```
Comment (peppy) [inline osu.Game/Tests/Visual/TestCaseEditor.cs]: maybe call this `EditorTestCase` in line with `OsuTestCase`?

Comment (smoogipoo) [reply osu.Game/Tests/Visual/TestCaseEditor.cs]: Fine with `TestCaseEditor : EditorTestCase` ?

Comment (peppy) [reply osu.Game/Tests/Visual/TestCaseEditor.cs]: yeah, just staying in line with the other abstract test case implementations

Comment (smoogipoo) [reply osu.Game/Tests/Visual/TestCaseEditor.cs]: cool

**Review by peppy — APPROVED** [review]

**Pull request merged**